src/services/Entries.php

```diff
@@ -952,16 +953,41 @@ private function _ensureSingleEntry(Section $section, ?array $siteSettings = nul
         // Get/save the entry with updated title, slug, and URI format
         // ---------------------------------------------------------------------
 
+        $baseEntryQuery = Entry::find()
+            ->sectionId($section->id)
+            ->siteId($siteIds)
+            ->status(null);
+
         // If there are any existing entries, find the first one with a valid typeId
         /** @var Entry|null $entry */
-        $entry = Entry::find()
+        $entry = $baseEntryQuery
             ->typeId($entryTypeIds)
-            ->sectionId($section->id)
-            ->siteId($siteIds)
-            ->status(null)
```

Comment:
This seems pretty important – at least `sectionId`. We wouldn’t want to fetch an entry from a different section that happens to share the same entry type here.

Reply:
Sure, but it’s still being used - I only moved it up a few lines into a variable.